src/shared/sleep-config.c

```diff
@@ -102,6 +102,47 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* If battery percentage capacity is less than equal to 5% return success */
+int battery_is_low(void) {
+        int r;
+
+        r = read_battery_capacity_percentage();
+        if (r < 0)
+               return r;
```

Comment:
https://github.com/systemd/systemd/pull/23640#discussion_r907302079
Am i getting confused with what you said here....
that retunr negative err in such case.
Although your suggestion is right, ideally it should be only 0 or 1 and nothing else.

Reply:
I don't think the error should be hidden, low battery or no battery are quite different and need to be handled differently

Reply:
Also the case of no battery is different from low battery so it should be handled separately.

Reply:
Sorry for the confusion, but these two comments are not contradict with each other. That is, expected and safe errors should be gracefully handled in the function, but any unexpected errors should be propagated:
```
int battery_is_low() {
    r = read_something();
    if (r == -ENOENT)  /* expected and safe error. */
        return false;
    if (r < 0) /* unexpected errors */
        return r;
    ....
}
```

Reply:
So when we check if battery_is_low > 0, i think in case of no battery while will not be invoked and system will directly hibernate

Reply:
woken_by_timer = FLAGS_SET(r, POLLIN);

                r = read_battery_capacity_percentage();
                if (r > 0) {
                        current_capacity = r;
                        log_debug("Current battery charge percentage after wakeup: %d%%", current_capacity);
                } else if (r == -ENOENT) {
                        log_debug_errno(r, "Battery capacity percentage unavailable, cannot estimate discharge rate: %m");
                        break;
                        /* In case of no battery, system will be hibernated */
I don't think we will be stuck in loop. 
Think about this, if we allow battery_is_low to return false in case of no battery i.e. -ENOENT and then system is suspended for hibernatedelaysec duration and on wakeup read_battery_capacity_percentage will return -ENOENT and then when break() will be encountered, system will hibernate